alica_engine/include/engine/AlicaContext.h

```diff
@@ -497,9 +501,11 @@ void AlicaContext::addSolver(Args&&... args)
 {
     static_assert(std::is_base_of<ISolverBase, SolverType>::value, "Must be derived from ISolverBase");
 #if (defined __cplusplus && __cplusplus >= 201402L)
-    _solvers.emplace(typeid(SolverType).hash_code(), std::make_unique<SolverType>(_engine.get(), std::forward<Args>(args)...));
+    _solvers.emplace(
+            typeid(SolverType).hash_code(), std::make_unique<SolverType>(editBlackboard(), getResultStore(), _configRootNode, std::forward<Args>(args)...));
```

Comment:
can't we do `_engine.editBlackboard` & `_engine.getResultStore` here?

Reply:
Can we also add a check here to verify if the engine is initialized? Otherwise it will crash. 

Its a bit strange that unlike the other add<> methods, the engine has to be initialized to add solvers. Perhaps this dependency can be removed in a subsequent PR by moving Blackboard to the alica context & getting rid of the dependency of VariableSyncModule on TeamManager. Seems like the only thing that is used from the team manager is checking if an agent is ignored & no one seems to be using this API at the moment.